### PR TITLE
[CADC-13816] Implement linting and style checks

### PR DESCRIPTION
**Pull request by shinybrar**
- Changed style format from `cadc-quality` to `palantir-java-format`
- Changed formatting / lint engine from `checkStyle` to `spotless`
- Added checking for `yaml|java|gradle` file formats.
- Fixed excessive token permission issues for codeQL action
- Changed CodeQL / OpenSSF Scorecard actions to run at midnight each night on schedule.

### CODE REVIEW

File: skaha/build.gradle [hunk 1]
```diff
@@ -1,7 +1,7 @@
 plugins {
     id 'war'
     id 'maven-publish'
-    id 'checkstyle'
+    id("com.diffplug.spotless") version "6.25.0"
```
Comment (at88mph) [inline skaha/build.gradle]: Just for consistency, please match this to the others:
```groovy
id 'com.diffplug.spotless' version '6.25.0'
```

**Review by at88mph — CHANGES_REQUESTED** [review]

**Review by at88mph — APPROVED** [review]

File: .github/workflows/ci.linting.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+jobs:
+  spotless:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.2.2
+        with:
+            fetch-depth: 0
+      - name: Setup Java
+        uses: actions/setup-java@v2
```
Comment (at88mph) [inline .github/workflows/ci.linting.yml]: This is at version 4 as well.  Please bump it.

File: .github/workflows/ci.testing.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.2.2
+        with:
+            fetch-depth: 0
+      - name: Setup Java
+        uses: actions/setup-java@v2
```
Comment (at88mph) [inline .github/workflows/ci.testing.yml]: This is at version 4 as well. Please bump it.

**Review by at88mph — CHANGES_REQUESTED** [review]

**Review by at88mph — DISMISSED** [review]

**Review by at88mph — DISMISSED** [review]

**Comment by shinybrar** [comment]
Fixes CADC-13811, CADC-13812 and CADC-13816

File: skaha/opencadc.gradle [hunk 4]
```diff
@@ -1,21 +1,7 @@
 configurations {
-    checkstyleDep
-    intTestImplementation.extendsFrom testImplementation
```
Comment (at88mph) [inline skaha/opencadc.gradle:3]: The `intTestImplementation` still needs to be there to run the integration tests.

**Review by at88mph — CHANGES_REQUESTED** [review]

**Review by at88mph — APPROVED** [review]

**Pull request merged**